packages/@eventual/core-runtime/src/local/local-persistance-store.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  Dirent,
+  mkdirSync,
+  readFileSync,
+  readdirSync,
+  rmSync,
+  statSync,
+  writeFileSync,
+} from "fs";
+import path from "path";
+
+export interface LocalSerializable {
+  serialize(): Record<string, Buffer>;
+}
+
+/**
+ * A local persistance store for local runtime data.
+ *
+ * Intentionally synchronous to support constructors and ensure a consistent point in time when saving.
```

Comment:
This makes sense. But are we also worried about performance? Fast hot reload is important

Reply:
What's the consistency problem? Can't we solve that with sequencing in the boot?